File: src/main/kotlin/cz/uhk/ppro/mhjp/resftulgallery/dto/GalleryDto.kt
package cz.uhk.ppro.mhjp.resftulgallery.dto

class SubmitGalleryDto(
        val title: String = "",
        val private: Boolean = false,
        val images: List<String> = emptyList()
)

class AddRemoveImagesFromGalleryDto(
        val images: List<String> = emptyList()
)

open class GalleryDataDto(
        val uuid: String,
        val title: String,
        val private: Boolean,
        val author: ListUserDataDto,
        val images: List<DataDto>
) : DataDto()

open class ListGalleryDataDto(
        val uuid: String
) : DataDto()

File: src/main/bobril/cz/uhk/ppro/mhjp/gallery/api/dto/galleryDto.ts
import {DataDto} from "./baseDto";
import {ListUserDataDto} from "./UserDto";

export interface SubmitGalleryDto {
        title: string;
        private: boolean;
        images: string[]
}

export interface AddRemoveImagesFromGalleryDto{
        images: string[]
}

export interface GalleryDataDto extends DataDto {
        uuid: string,
        title: string,
        private: boolean,
        author: ListUserDataDto,
        images: DataDto[]
}

export interface ListGalleryDataDto extends DataDto {
        uuid: string
}

